File: app/src/main/java/cz/kutner/comicsdb/fragment/SeriesFragment.java
package cz.kutner.comicsdb.fragment;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.google.android.gms.analytics.HitBuilders;
import com.google.android.gms.analytics.Tracker;

import cz.kutner.comicsdb.ComicsDBApplication;
import cz.kutner.comicsdb.connector.SeriesConnector;
import cz.kutner.comicsdb.holder.SeriesViewHolder;
import cz.kutner.comicsdb.model.Series;
import rx.Observable;
import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;
import uk.co.ribot.easyadapter.EasyRecyclerAdapter;

public class SeriesFragment extends AbstractFragment<Series> {


    public SeriesFragment() {
        super();
        preloadCount = 20;
    }

    public static SeriesFragment newInstance() {

        Bundle args = new Bundle();

        SeriesFragment fragment = new SeriesFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        adapter = new EasyRecyclerAdapter<Series>(
                getContext(),
                SeriesViewHolder.class,
                data);
        return super.onCreateView(inflater, container, savedInstanceState);
    }

    void loadData() {
        if (!searchRunning) {
            searchRunning = true;
            Observable.just(lastPage)
                    .subscribeOn(Schedulers.io())
                    .map(integer -> SeriesConnector.get(integer))
                    .observeOn(AndroidSchedulers.mainThread())
                    .subscribe(series -> {
                        result = series;
                        showData();
                    });
            lastPage++;
        }
    }

    @Override
    public void onStart() {
        super.onStart();
        Bundle args = this.getArguments();
        Tracker tracker = ComicsDBApplication.getTracker();
        ((AppCompatActivity) getActivity()).getSupportActionBar().setTitle("Serie");
        tracker.setScreenName("SeriesFragment");
        tracker.send(new HitBuilders.ScreenViewBuilder().build());
    }
}

File: app/src/main/kotlin/cz/kutner/comicsdb/fragment/SeriesFragment.kt
package cz.kutner.comicsdb.fragment

import android.app.SearchManager
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import cz.kutner.comicsdb.adapter.SeriesListAdapter
import cz.kutner.comicsdb.model.Series
import cz.kutner.comicsdb.utils.logVisit
import java.text.Normalizer

class SeriesFragment : AbstractFragment<Series>() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        adapter = SeriesListAdapter(inflater, data)
        return super.onCreateView(inflater, container, savedInstanceState)
    }

    override fun loadData() {
        if (!searchRunning) {
            searchRunning = true
            val args = this.arguments
            var searchText = ""
            if (args != null && args.containsKey(SearchManager.QUERY)) {
                searchText = args.getString(SearchManager.QUERY)
                searchText = Normalizer.normalize(searchText, Normalizer.Form.NFD).replace("[\\p{InCombiningDiacriticalMarks}]".toRegex(), "")
            }
            val call = retrofitModule.seriesListService.getSeriesList(lastPage * preloadCount, preloadCount, searchText)
            runAsync(call)
        }
    }

    override fun onStart() {
        super.onStart()
        val args = this.arguments
        if (args != null && args.containsKey(SearchManager.QUERY)) {
            (activity as AppCompatActivity).supportActionBar?.title = "Výsledek pro \"" + args.getString(SearchManager.QUERY) + "\""
        } else {
            (activity as AppCompatActivity).supportActionBar?.title = "Serie"
            firebase.logVisit(contentName = "Zobrazení seznamu sérií", contentType = "Série")
        }
    }

    companion object {

        fun newInstance(): SeriesFragment {

            val args = Bundle()

            val fragment = SeriesFragment()
            fragment.arguments = args
            return fragment
        }
    }
}


